web/src/js/components/Settings/SettingsPageComponent.js

```diff
@@ -120,11 +127,73 @@ class SettingsPage extends React.Component {
             justifyContent: 'center',
           }}
         >
-          {React.Children.map(this.props.children, child =>
-            React.cloneElement(child, {
-              showError: showError.bind(this),
-            })
-          )}
+          <Switch>
+            <Route
+              exact
+              path="/newtab/settings"
+              render={props => (
+                <WidgetsSettingsView
+                  {...props}
+                  showError={showError.bind(this)}
+                />
+              )}
+            />
+            <Route
+              exact
+              path="/newtab/settings/widgets"
+              render={props => (
+                <WidgetsSettingsView
+                  {...props}
+                  showError={showError.bind(this)}
+                />
+              )}
+            />
+
```

Comment:
Make a non-exact /newtab/profile route at the bottom of the Switch so we fall back to this route for any /newtab/profile/* path.